### PR TITLE
[TensorDescriptor] Fix API description, add const for input data pointers

### PR DESCRIPTION
**Pull request by averinevg**
- Incorrect direction of input/output parameters in the description has been fixed
- `const` type qualifier has been added for input data pointers

### CODE REVIEW

File: include/miopen/miopen.h [hunk 1]
```diff
@@ -592,7 +592,7 @@ MIOPEN_EXPORT miopenStatus_t miopenCreateTensorDescriptor(miopenTensorDescriptor
  *
  * Interface for setting 4-D tensor shape. MIOpen currently only implements NCHW layout.
  *
- * @param tensorDesc Tensor descriptor type (output)
+ * @param tensorDesc Tensor descriptor (input)
```
Comment (atamazov) [inline include/miopen/miopen.h]: Formally, this is an input parameter, but given that this call modifies the internal state of the existing descriptor, the parameter is actually input/output.

File: include/miopen/miopen.h [hunk 2]
```diff
@@ -606,7 +606,7 @@ MIOPEN_EXPORT miopenStatus_t miopenSet4dTensorDescriptor(
 /*! @brief Set shape of ND tensor with specific layout
  *
  * Interface for setting N-D tensor shape. This interface support NHWC, NCHW, NCHWc*, CHWNc*
- * @param tensorDesc   Tensor descriptor type (output)
+ * @param tensorDesc   Tensor descriptor (input)
```
Comment (atamazov) [inline include/miopen/miopen.h]: Ditto

File: include/miopen/miopen.h [hunk 3]
```diff
@@ -617,13 +617,13 @@ MIOPEN_EXPORT miopenStatus_t
...
                                       miopenTensorLayout_t tensorLayout,
-                                      int* lens,
+                                      const int* lens,
                                       int num_lens);
 /*! @brief Set shape and stride of 4D tensor
  *
  * Interface for setting 4-D tensor shape and stride.
  *
- * @param tensorDesc Tensor descriptor type (output)
+ * @param tensorDesc Tensor descriptor (input)
```
Comment (atamazov) [inline include/miopen/miopen.h]: Ditto

File: include/miopen/miopen.h [hunk 4]
```diff
@@ -677,7 +677,7 @@ MIOPEN_EXPORT miopenStatus_t miopenGet4dTensorDescriptor(miopenTensorDescriptor_
  *
  * Interface for setting tensor shape. MIOpen has support for 1, 2, 3, 4, 5 dimensional tensor of
  * layout.
- * @param tensorDesc   Tensor descriptor type (input)
+ * @param tensorDesc   Tensor descriptor (input)
```
Comment (atamazov) [inline include/miopen/miopen.h]: Ditto

**Review by atamazov — CHANGES_REQUESTED** [review]

**Comment by atamazov** [comment]
[Informative] While this PR changes signatures of some API calls, these changes are non-breaking. Therefore I am not insisting on adding [API] to the title or labeling with https://github.com/ROCmSoftwarePlatform/MIOpen/labels/IMPACTS_API.

File: include/miopen/miopen.h (hunk 1 above)
Comment (averinevg) [reply include/miopen/miopen.h]: The description has been fixed

File: include/miopen/miopen.h (hunk 2 above)
Comment (averinevg) [reply include/miopen/miopen.h]: The description has been fixed

File: include/miopen/miopen.h (hunk 3 above)
Comment (averinevg) [reply include/miopen/miopen.h]: The description has been fixed

File: include/miopen/miopen.h (hunk 4 above)
Comment (averinevg) [reply include/miopen/miopen.h]: The description has been fixed

**Review by atamazov — APPROVED** [review]
LGTM

File: include/miopen/miopen.h (hunk 1 above)
Comment (atamazov) [reply include/miopen/miopen.h]: [Notice] @averinevg In trivial cases, please mark fixed review comments as _resolved_ thus hiding them (I can't do that)

**Review by junliume — APPROVED** [review]

**Pull request merged**